### ISSUE TITLE
t.words.basesupport attributeerror when "already connecting"

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2086]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| moonfallen reported|
|-|-|
|Trac ID|trac#2086|
|Type|defect|
|Created|2006-09-13 15:30:41Z|

When you attempt to connect a basesupport.AbstractAccount twice, you get an AttributeError, because it attempts to use t.i.error.ConnectionError.  The correct exception is t.i.error.ConnectError.

Attaching patch with unit test.

Attachments:

* [basesupport.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/aa2/aa23f7c48a6074023875b7c69a7f4873243ebd36/87a72a48f3015c4abf02732976c8f7eaf2d9d500.patch) (1315 bytes) - added by moonfallen on 2006-09-13 15:30:56Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2086 2086
type__defect defect
reporter__moonfallen moonfallen
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1158161441000000 1158161441000000
changetime__1158904337000000 1158904337000000
version__None None
owner__moonfallen moonfallen
cc__corydodt@...
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_1"></a>|@jerub set owner to moonfallen|
|-|-|

I created a branch, irc-account-2086, and played around with your patch.

It turns out that the callback chain in .logOn is broken, and will end up calling both _loginFailed and _cb_logOn.

Please see the unit tests I've written in twisted.words.test.test_basesupport.

If you can fix .logOn so it has the expected behaviour (fix the failing unit test), I'd appreciate it. Do you svn commit access? Got Combinator set up?

Mark it back to review and give it back to me when you're done.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|moonfallen commented|
|-|-|

Thanks, I'll take a look.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|moonfallen set owner to @jerub|
|-|-|

OK, committed r18119 on the branch.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|moonfallen removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set owner to moonfallen|
|-|-|

Thanks for adding more tests than strictly necessary for this branch. 

A few points:
 * `return reason` works just as well as `reason.raiseException()`, and better matches the docstring.
 * test_basesupport.py:61,78,79 are all > 80 columns
 * Please add a docstring for test_alreadyConnecting

Please fix all of these things, then merge.

**Comment by twisted-trac** [comment]
|<img alt="moonfallen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|moonfallen set status to `closed`|
|-|-|

(In [18227]) Author: jerub, moonfallen
Reviewer: jml
Fixes [#2086](https://github.com/twisted/twisted/issues/2086)

Use the right name for the exception, and increase by infinity percent the
number of tests for basesupport.
